### DISCUSSION TITLE
It looks like you're trying to use TypeScript but do not have the required package(s) installed

### DISCUSSION THREAD

**Discussion by razznblue** [vercel/community#4373] [Help] [answered]
### Summary

Hello!

I am trying to deploy my next.js app to vercel. And it is not recognizing typescript. I have all the dependencies installed that the error is saying I don't have.

![Screen Shot 2023-10-05 at 1 08 04 PM](https://github.com/vercel/community/assets/59562158/08cffadb-8aee-4c8e-bbe9-5b366e3eab29)

I've done my research and discovered others with my same issue. They have found that downgrading their @types/react version was the solution but that does not work for me. I think it is because that solution is outdated as it is from early 2022.
https://stackoverflow.com/questions/71842787/next-js-typescript-error-you-do-not-have-the-required-packages-installed

Does anyone have any ideas or know of a solution?
Let me know If I can provide any additional information that might be helpful!

Thank you

### Steps to Reproduce

package.json
```
{
  "name": "test-app",
  "version": "0.1.0",
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@typegoose/typegoose": "^11.4.0",
    "axios": "^1.4.0",
    "jsonwebtoken": "^9.0.2",
    "mongoose": "^7.4.1",
    "next": "^13.5.3",
    "next-auth": "^4.22.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "winston": "^3.10.0"
  },
  "devDependencies": {
    "@types/jsonwebtoken": "^9.0.3",
    "@types/node": "20.4.2",
    "@types/react": "18.2.0",
    "@types/react-dom": "18.2.0",
    "eslint": "^8.50.0",
    "eslint-config-next": "^13.4.12",
    "typescript": "5.2.2"
  },
  "resolutions": {
    "@types/react": "^18.0.1"
  }
}
```

tsconfig.json
```
{
  "compilerOptions": {
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "plugins": [
      {
        "name": "next"
      }
    ],
    "strictNullChecks": true,
    "strictPropertyInitialization": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true
  },
  "include": [
    "next-env.d.ts",
    ".next/types/**/*.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}

```

**Comment by amyegan** [comment]
Can you share [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use to help with debugging?

**Comment by razznblue** [comment]
@amyegan - thank you for your response.
I was only able to get this working by deleting my vercel project and starting over from scratch.
However, this error came back today. I did not change anything in package.json except adding cheerio as a dependency.

I cannot reproduce the error locally except by using the vercel cli by running `vercel build` that gives me the same error.

I am using these versions of the libraries in the error:
  "@types/react": "18.2.15",
  "typescript": "5.1.6"

The packages are in my package.json, so why isn't vercel recognizing them?
Thanks

**Comment by razznblue** [comment, marked as answer]
Update:

My issue was that in vercel I was setting the NODE_ENV environment variable. You are not supposed to do that because next automatically adds this env when building the application. After removing NODE_ENV, my application deployed! This also applies in local development too.

In theory, I figured there should be a better error message, but that may be out of their hands. I'm not sure how that all works.
